### DISCUSSION TITLE
Help with appending data from external files with jQuery

### DISCUSSION THREAD

**Discussion by Ethan-Mircheff** [community/community#146867] [Programming Help] [answered]
### Body

I am trying to make a Github pages site that appends the data from several HTML files with jQuery (the  files are small now, but they will be a lot bigger in the final version), and it works other than one issue. Right now, there is a gap in the text between the two sources, but what I would like to do is make a single string of all the text so it can be copied and pasted.
<br>
Main code:
```
<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>One Billion Digits of Pi</title>
  <style>
  body {
    font-size: 20px;
    font-family: "Trebuchet MS", Veranda, sans-serif;
    display: inline-block;
    margin: 10px;
  }
    
  .digits {
    display: inline-block
  }
  </style>
  <script src="https://code.jquery.com/jquery-3.7.1.js"></script>
</head>
<body>

<h1>One Billion Digits of Pi:</h1>
  <p id="digits"></p>
    
<script>
  $(document).ready(function() {
  var paragraph = $("#digits");
  var loadFiles = ["digits.html","digits2.html"];
  $.each(loadFiles, function(index, file) {
    $.get(file, function(data) {
      paragraph.append(data);
    });
  });
});
</script>
</body>
</html>
``` 
<br>
One of the appended files:

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
   <style>
  body {
    font-size: 20px;
    font-family: Serif;
  }
  </style>
</head>
  <body>
    3.141592653589793238462643382
  </body>
</html>
```
<br>

[The site on Github pages](https://ethan-mircheff.github.io/pibillion/)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MindfulLearner** [comment]
try this 

```js
$(document).ready(function() {
  var paragraph = $("#digits");
  var loadFiles = ["digits.html", "digits2.html"];
  $.each(loadFiles, function(index, file) {
    $.get(file, function(data) {
    
      var tempDiv = $("<div>").html(data);
      var textContent = tempDiv.text();
      
      paragraph.append(textContent);
    });
  });
});
```

added: 
```js
      var tempDiv = $("<div>").html(data);
      var textContent = tempDiv.text();
      ```

**Reply by Ethan-Mircheff** [reply]
I think that's almost there, but I'm getting a new issue now. For some reason it is now also appending some of the formating information into the text on the [page](https://ethan-mircheff.github.io/pibillion/).
![Screenshot 2024-12-13 7 37 11 AM](https://github.com/user-attachments/assets/fdfdd640-5ddf-405a-ac5f-025d4964b953)
Do I just need to modify a div? I'm still fairly new to JavaScript.

**Reply by Ethan-Mircheff** [reply]
I solved this by getting rid of the unnecessary styling in the files holding the digits, but there is still a space between the two appended strings of text.
![Screenshot 2024-12-13 12 36 50 PM](https://github.com/user-attachments/assets/58bb70af-499b-45d1-8fe3-4fcbba89feb6)
Is there something else that I could try to remove the space?

**Reply by MindfulLearner** [reply]
implode or explode?

**Reply by Ethan-Mircheff** [reply]
I could try to use implode, could you give me an example of how to implement that in my code? I haven't ever used php code before, so  I'm not really sure how

**Reply by Ethan-Mircheff** [reply, marked as answer]
I figured it out! I tried one more time with JavaScript, and I added some code to remove the space:
```html
<script>
  $(document).ready(function() {
  var paragraph = $("#digits");
  var loadFiles = ["digits.html","digits2.html"];
  $.each(loadFiles, function(index, file) {
    $.get(file, function(data) {

      var tempDiv = $("<div>").html(data);
      var textContent = tempDiv.text();
      var textNoSpaces = textContent.replace(/\s+/g, "");
      
      paragraph.append(textNoSpaces);
    });
  });
});
</script>
```
<br>
So the final full code is:
<br>

```html
<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>One Billion Digits of Pi</title>
  <style>
  body {
    font-size: 20px;
    font-family: "Trebuchet MS", Veranda, sans-serif;
    display: inline-block;
    margin: 10px;
  }
    
  .digits {
    display: inline-block;
    font-family: serif;
  }
  </style>
  <script src="https://code.jquery.com/jquery-3.7.1.js"></script>
</head>
<body>

<h1>One Billion Digits of Pi:</h1>
  <p id="digits"></p>
  
<script>
  $(document).ready(function() {
  var paragraph = $("#digits");
  var loadFiles = ["digits.html","digits2.html"];
  $.each(loadFiles, function(index, file) {
    $.get(file, function(data) {

      var tempDiv = $("<div>").html(data);
      var textContent = tempDiv.text();
      var textNoSpaces = textContent.replace(/\s+/g, "");
      
      paragraph.append(textNoSpaces);
    });
  });
});
</script>
</body>
</html>
```
<br>
Thanks for the help!
